bofire/data_models/constraints/constraint.py

```diff
@@ -12,7 +12,7 @@
 class Constraint(BaseModel):
     """Abstract base class to define constraints on the optimization space."""
 
-    type: Any
+    type: str
```

Comment:
It has to by `Any`, in the abstact base class. Since it is always `Literal` in the derived class.